modules/core/src/main/java/org/locationtech/jts/coverage/CoverageEdge.java

```diff
@@ -26,12 +26,16 @@ class CoverageEdge {
 
   public static CoverageEdge createEdge(LinearRing ring) {
     Coordinate[] pts = extractEdgePoints(ring, 0, ring.getNumPoints() - 1);
-    return new CoverageEdge(pts);
+    CoverageEdge edge = new CoverageEdge(pts);
+    edge.nodeCount = 0;
+    return edge;
   }
 
   public static CoverageEdge createEdge(LinearRing ring, int start, int end) {
     Coordinate[] pts = extractEdgePoints(ring, start, end);
-    return new CoverageEdge(pts);
+    CoverageEdge edge = new CoverageEdge(pts);
+    edge.nodeCount = start == end ? 1 : 2;
```

Comment:
This is only used to check if it is zero, so can it just be a boolean (say `isNodePreserved`) ?